### DISCUSSION TITLE
Enabling disabled NodeJs package (crypto)

### DISCUSSION THREAD

**Discussion by itsmeyaw** [eclipse-theia/theia#13545] [Q&A] [answered]
Hello, I plan to use the crypto module from NodeJs for my front-end contribution. However, Theia Webpack configuration does not allow me to resolve it (as seen in the image below in line 148).

![image](https://github.com/eclipse-theia/theia/assets/10561001/df1644c5-e2d2-418b-ae9a-28f60e2f013a)

Since I should not edit this generated webpack configuration, is there a way to enable the crypto package for NodeJs in the build? 

Edit: Actually, my error is `root ERROR Could not start contribution TypeError: crypto.randomBytes is not a function` and I checked that the configuration set crypto to false.

Thank you.

**Comment by msujew** [comment]
Hey @itsmeyaw,

there should be a `webpack.config.js` file in your app project. You are supposed to perform any changes of the webpack config in there. 

The only reason why the generated webpack config files aren't supposed to be edited is because they are regenerated on every build. Thus, you lose all changes you did.

**Reply by itsmeyaw** [reply]
Yes, I did tried to change it but still resulting **(Edit:) another error**. Below is my change (line 18-20):

![image](https://github.com/eclipse-theia/theia/assets/10561001/b863724f-9355-428c-af62-3c732486996f)

And the error is: `Cannot find module 'crypto'`

Do you know what is wrong here?

**Comment by itsmeyaw** [comment, marked as answer]
Adding to answer from @msujew, I just realized that you need to install and use the "browserified" packages for that. Hence, you need to add these:

![image](https://github.com/eclipse-theia/theia/assets/10561001/e4a5eefd-1b43-49cf-b12f-f15ebe7c0f0c)

and install the aforementioned npm packages.
